In [ ]:
import numpy as np
import pandas as pd

In [ ]:
dataset=pd.read_csv("/content/DataCoSupplyChainDataset.csv.csv",encoding="unicode_escape")
import pandas as pd
dataset

Type  ...   Shipping Mode
0          DEBIT  ...  Standard Class
1       TRANSFER  ...  Standard Class
2           CASH  ...  Standard Class
3          DEBIT  ...  Standard Class
4        PAYMENT  ...  Standard Class
...          ...  ...             ...
180514      CASH  ...  Standard Class
180515     DEBIT  ...    Second Class
180516  TRANSFER  ...  Standard Class
180517   PAYMENT  ...  Standard Class
180518   PAYMENT  ...  Standard Class

[180519 rows x 53 columns]

## List all the cities and the respective number of pending orders in "India"?

In [ ]:
dataset_1 = dataset.iloc[:,[25,26,42]]
dataset_1

Order City Order Country     Order Status
0           Bekasi     Indonesia         COMPLETE
1          Bikaner         India          PENDING
2          Bikaner         India           CLOSED
3       Townsville     Australia         COMPLETE
4       Townsville     Australia  PENDING_PAYMENT
...            ...           ...              ...
180514    Shanghái         China           CLOSED
180515    Hirakata         Japón         COMPLETE
180516    Adelaide     Australia          PENDING
180517    Adelaide     Australia  PENDING_PAYMENT
180518   Nagercoil         India  PENDING_PAYMENT

[180519 rows x 3 columns]

In [ ]:
count=dataset_1['Order Status'].value_counts() 
count

COMPLETE           59491
PENDING_PAYMENT    39832
PROCESSING         21902
PENDING            20227
CLOSED             19616
ON_HOLD             9804
SUSPECTED_FRAUD     4062
CANCELED            3692
PAYMENT_REVIEW      1893
Name: Order Status, dtype: int64

In [ ]:
dataset_1_filtered = dataset_1[dataset_1['Order Status'] == 'PENDING']
dataset_1_filtered

Order City Order Country Order Status
1         Bikaner         India      PENDING
11         Manado     Indonesia      PENDING
12         Manado     Indonesia      PENDING
21       Jabalpur         India      PENDING
22        Geelong     Australia      PENDING
...           ...           ...          ...
180414   Shenzhen         China      PENDING
180458   Semarang     Indonesia      PENDING
180485      Korla         China      PENDING
180491    Bendigo     Australia      PENDING
180516   Adelaide     Australia      PENDING

[20227 rows x 3 columns]

In [ ]:
dataset_1_filtered_1 = dataset_1_filtered[dataset_1_filtered['Order Country'] == 'India']
dataset_1_filtered_1

Order City Order Country Order Status
1           Bikaner         India      PENDING
21         Jabalpur         India      PENDING
45        Bangalore         India      PENDING
46        Bangalore         India      PENDING
382            Pune         India      PENDING
...             ...           ...          ...
180142  Bhubaneswar         India      PENDING
180164         Agra         India      PENDING
180248        Dewas         India      PENDING
180331        Ajmer         India      PENDING
180405        Patna         India      PENDING

[537 rows x 3 columns]

In [ ]:
count=dataset_1_filtered_1['Order City'].value_counts() 
count

Kanpur       25
Bangalore    24
Pune         24
Lucknow      21
Delhi        18
             ..
Warangal      1
Vadodara      1
Tiruppur      1
Hubli         1
Madurai       1
Name: Order City, Length: 91, dtype: int64

Therefore above is the number of pending orders in India

## Which country has the most suspected fraud?

In [ ]:
dataset_2 = dataset.iloc[:,[26,42]]
dataset_2

Order Country     Order Status
0          Indonesia         COMPLETE
1              India          PENDING
2              India           CLOSED
3          Australia         COMPLETE
4          Australia  PENDING_PAYMENT
...              ...              ...
180514         China           CLOSED
180515         Japón         COMPLETE
180516     Australia          PENDING
180517     Australia  PENDING_PAYMENT
180518         India  PENDING_PAYMENT

[180519 rows x 2 columns]

In [ ]:
dataset_2_filtered = dataset_2[dataset_2['Order Status'] == 'SUSPECTED_FRAUD']
dataset_2_filtered

Order Country     Order Status
10               Japón  SUSPECTED_FRAUD
24           Australia  SUSPECTED_FRAUD
183     Estados Unidos  SUSPECTED_FRAUD
184     Estados Unidos  SUSPECTED_FRAUD
185     Estados Unidos  SUSPECTED_FRAUD
...                ...              ...
180274           Japón  SUSPECTED_FRAUD
180309           China  SUSPECTED_FRAUD
180352       Indonesia  SUSPECTED_FRAUD
180406           India  SUSPECTED_FRAUD
180490           India  SUSPECTED_FRAUD

[4062 rows x 2 columns]

In [ ]:
count=dataset_2_filtered['Order Country'].value_counts() 
count

Estados Unidos     573
Francia            362
México             317
Alemania           227
Australia          197
                  ... 
Costa de Marfil      1
Georgia              1
Benín                1
Uzbekistán           1
Grecia               1
Name: Order Country, Length: 102, dtype: int64

## Estados Unidos has the most suspected fraud of 573

## Which product(s) has/have the maximum discount?

In [ ]:
dataset_3 = dataset.iloc[:,[48,31]]
dataset_3

Product Name  Order Item Discount
0                                    Smart watch             13.110000
1                                    Smart watch             16.389999
2                                    Smart watch             18.030001
3                                    Smart watch             22.940001
4                                    Smart watch             29.500000
...                                           ...                  ...
180514  Field & Stream Sportsman 16 Gun Fire Safe             0.000000
180515  Field & Stream Sportsman 16 Gun Fire Safe             4.000000
180516  Field & Stream Sportsman 16 Gun Fire Safe             8.000000
180517  Field & Stream Sportsman 16 Gun Fire Safe            12.000000
180518  Field & Stream Sportsman 16 Gun Fire Safe            16.000000

[180519 rows x 2 columns]

In [ ]:
maxValueIndex = dataset_3['Order Item Discount'].idxmax(axis = 0)
print("Max value Sno.:")
maxValueIndex

Max value Sno.:


12606

In [ ]:
max = dataset_3['Order Item Discount'].max()
print("Maximum value in column 'x': " )
print(max , dataset_3.loc[dataset_3['Order Item Discount']== max])

Maximum value in column 'x': 
500.0               Product Name  Order Item Discount
12606  SOLE E35 Elliptical                500.0


SOLE E35 Elliptical has the maximum discount of 500.0

## How many successfully processed orders have days for shipping real (Real) > Days for shipment (Scheduled)?

In [ ]:
dataset_4 = dataset.iloc[:,[1,2,42]]
dataset_4

Days for shipping (real)  ...     Order Status
0                              3  ...         COMPLETE
1                              5  ...          PENDING
2                              4  ...           CLOSED
3                              3  ...         COMPLETE
4                              2  ...  PENDING_PAYMENT
...                          ...  ...              ...
180514                         4  ...           CLOSED
180515                         3  ...         COMPLETE
180516                         5  ...          PENDING
180517                         3  ...  PENDING_PAYMENT
180518                         4  ...  PENDING_PAYMENT

[180519 rows x 3 columns]

In [ ]:
dataset_4_filtered = dataset_4[dataset_4['Order Status'] == 'COMPLETE']
dataset_4_filtered

Days for shipping (real)  Days for shipment (scheduled) Order Status
0                              3                              4     COMPLETE
3                              3                              4     COMPLETE
6                              2                              1     COMPLETE
13                             2                              1     COMPLETE
15                             2                              1     COMPLETE
...                          ...                            ...          ...
180500                         3                              4     COMPLETE
180505                         2                              4     COMPLETE
180508                         4                              2     COMPLETE
180512                         6                              4     COMPLETE
180515                         3                              2     COMPLETE

[59491 rows x 3 columns]

In [ ]:
dataset_4_filtered_1 = dataset_4_filtered[dataset_4_filtered['Days for shipping (real)'] > dataset_4_filtered['Days for shipment (scheduled)'] ]
dataset_4_filtered_1

Days for shipping (real)  Days for shipment (scheduled) Order Status
6                              2                              1     COMPLETE
13                             2                              1     COMPLETE
15                             2                              1     COMPLETE
18                             2                              1     COMPLETE
25                             6                              2     COMPLETE
...                          ...                            ...          ...
180479                         6                              4     COMPLETE
180499                         4                              2     COMPLETE
180508                         4                              2     COMPLETE
180512                         6                              4     COMPLETE
180515                         3                              2     COMPLETE

[34199 rows x 3 columns]

In [ ]:
count=dataset_4_filtered_1['Order Status'].value_counts() 
count

COMPLETE    34199
Name: Order Status, dtype: int64

## 34199 successfully processed orders have days for shipping real (Real) > Days for shipment (Scheduled)